### PR TITLE
Column function chaining alias

### LINKED ISSUE #7190: Can't refer to alias in function chaining syntax
### What happens?

When using an alias of a calculated column defined previously in the same select statement, I am unable to use the new function chaining / dot syntax.

### To Reproduce

```python
import duckdb
import pandas as pd

duckdb.sql("""
    select
        'test' || ' more testing' as added,
        added.substr(5) as my_substr
""").df()

---------------------------------------------------------------------------
CatalogException                          Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_67352/3026736190.py in <module>
----> 1 duckdb.sql("""
      2     select
      3         'test' || ' more testing' as added,
      4         added.substr(5) as my_substr
      5 """).df()

CatalogException: Catalog Error: Scalar Function with name substr does not exist!
Did you mean "main.substr"?

```

### OS:

Windows

### DuckDB Version:

0.7.2-dev2366

### DuckDB Client:

Python

### Full Name:

Alex Monahan

### Affiliation:

Intel and DuckDB Labs

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

### PR DESCRIPTION
**Pull request by douenergy**
Fix https://github.com/duckdb/duckdb/issues/7190

Now we can support this ! 

```sql
SELECT
        'test' || ' more testing' as added,
        added.substr(5) as my_substr,
        my_substr.upper() usubstr, 
        my_substr.lower() lsubstr,
        usubstr ||  lsubstr mix_substr
```
```
┌───────────────────┬───────────────┬───────────────┬───────────────┬────────────────────────────┐
│       added       │   my_substr   │    usubstr    │    lsubstr    │         mix_substr         │
│      varchar      │    varchar    │    varchar    │    varchar    │          varchar           │
├───────────────────┼───────────────┼───────────────┼───────────────┼────────────────────────────┤
│ test more testing │  more testing │  MORE TESTING │  more testing │  MORE TESTING more testing │
└───────────────────┴───────────────┴───────────────┴───────────────┴────────────────────────────┘
```

### CODE REVIEW

File: src/planner/binder/expression/bind_columnref_expression.cpp [hunk 1]
```diff
@@ -388,4 +388,10 @@ BindResult ExpressionBinder::BindExpression(ColumnRefExpression &colref_p, idx_t
 	return result;
 }
 
+bool ExpressionBinder::QualifyColumnAlias(const ColumnRefExpression &colref) {
+	// Only BaseSelectBinder will have a valid col alias map,
+	// otherwise just return false
```
Comment (Tishj) [inline src/planner/binder/expression/bind_columnref_expression.cpp:393]: Shouldn't we use the parent binder in this case?

Comment (douenergy) [reply src/planner/binder/expression/bind_columnref_expression.cpp:393]: I think only `BaseSelectBinder` has the alias_map we need to check
There is no alias_map data member in `ExpressionBinder` (parent class)

**Comment by douenergy** [comment]
Java JDBC CI pass on my fork.
https://github.com/douenergy/duckdb/actions/runs/4851730192

**Comment by Mytherin** [comment]
Thanks for the PR! Looks good to me.

**Comment by nicku33** [comment]
Can I just say how much I love this feature.

**Pull request merged**